graphium/data/multilevel_utils.py

```diff
@@ -39,6 +40,17 @@ def unpack_column(data: pd.Series):
     def merge_columns(data: pd.Series):
         data = data.to_list()
         data = [np.array([np.nan]) if not isinstance(d, np.ndarray) and math.isnan(d) else d for d in data]
+
+        # Ensure that all shapes are either the same, or 1
+        sizes = [d.shape[0] for d in data]
+        maxsize = max(sizes)
+        for size in sizes:
+            if (size != maxsize) and (size != 1):
```

Comment:
I assume the `size != 1` catches the nan-array created in line 42? It would probably be safer to specificially check for the nan-array. Let me think of a solution.